hashes/src/util.rs

```diff
@@ -222,9 +271,77 @@ macro_rules! hash_newtype {
                 &self.0[index]
             }
         }
+        )+
+    };
+}
+
```

Comment:
The `hash_newtype_struct` macro took me a while to work out. Perhaps throw a comment on it.
```
/// Declares the `$newtype` struct, the recursive macro calls are to strip the `#[hash_newtype(direction)]` attribute from the attributes list.
```

Reply:
Yeah, I guess I should document my design somewhere so that if people attempt "simpler" design they don't get into insane hair pulling like I did.

Reply:
Yeah my thought process while reading was "why the f*** did he do that ... oh for that reason, ouch that would have hurt to work out" :)